vllm/ex/ex.py

```diff
@@ -0,0 +1,245 @@
+import copy
+import torch
+import unittest.mock
+
+from .code_cache import CodeCache
+from .fusion import FusedOpGenerator, pointwise_fusion
+from .register import SUPPORTED
+from .rewrite_quantized_gemms import rewrite_quantized_gemms
+from .utils import ModuleInputGenerator, graph_print_tabular, is_call, call_method_class
+
+from torch._dynamo import register_backend, lookup_backend
+from torch.fx.passes.operator_support import create_op_support
+from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner, Partition
+from torch.fx.passes.tools_common import get_node_target
+from torch.fx.passes.shape_prop import ShapeProp
+from torch._subclasses.fake_tensor import FakeTensorMode, FakeTensor
+
+from typing import List, Tuple, Any, Dict, Optional, Callable, Mapping, Set
+
+from vllm.logger import init_logger
+
+import traceback
+
+logger = init_logger(__name__)
+
+###############################################################################
+#
+# Module partitioning
+#
+###############################################################################
+
+"""
+A callback for the fx CapabilityBasedPartitioner.  Nodes that are "supported"
+are partitioned into new submodules.
+"""
+def is_node_supported(
+    submodules: Mapping[str, torch.nn.Module],
+    node: torch.fx.Node,
+) -> bool:
+    if is_call(node):
+        fn_name = get_node_target(submodules, node)
+        if not fn_name in SUPPORTED:
```

Comment:
Do you need to handle subclasses here as well? You handle them for the `node` below.

Reply:
I can comment this better but `fn_name` is either the name of a function or method.  If it's a method then the `SUPPORTED` map value is a set of the classes that method is supported on.  So if `fn_name` is not a key in the `SUPPORTED` dict, it's definitely not supported.